src/controllers/dlgprefcontroller.cpp

```diff
@@ -1019,13 +1041,15 @@ void DlgPrefController::slotShowMapping(std::shared_ptr<LegacyControllerMapping>
                         [this] { setDirty(true); });
             }
         }
+    }
 
-        if (settings.isEmpty()) {
-            m_ui.groupBoxSettings->setVisible(false);
-        } else {
-            m_ui.groupBoxSettings->setVisible(true);
-            setScrollSafeGuardForAllInputWidgets(m_ui.groupBoxSettings);
+    // Show or hide the settings tab based on the presence of settings
+    if (pMapping && !pMapping->getSettings().isEmpty()) {
+        if (m_ui.controllerTabs->indexOf(m_ui.settingsTab) == -1) {
+            m_ui.controllerTabs->addTab(m_ui.settingsTab, tr("Mapping Settings"));
```

Comment:
This is obsolete with `settabVisible()`, so just FYI to avoid duplicate tr strings here and n ui file, we could use `QTabwidget::tabText(QTabwidget::indexOf(QWidget*))`